runtime/jvmti/jvmtiHook.c

```diff
@@ -2231,26 +2231,40 @@ jvmtiHookGCCycleEnd(J9HookInterface** hook, UDATA eventNum, void* eventData, voi
 }
 
 /**
- * Determines if the class described by <tt>classLoadData</tt> can 
+ * Determines if the class described by <tt>data</tt> can 
  * be instrumented.
- * @param classLoadData 
+ * @param data The class file load hook event data
  * @return TRUE if the class can be instrumented. 
  */
 static BOOLEAN
-canClassBeInstrumented(J9VMClassLoadHookEvent* classLoadData)
-{
-	BOOLEAN result = TRUE;
-
-	/* Skip any class that does not have a name specified */
-	if (classLoadData->className == NULL) {
-		result = FALSE;
-	} else if (strcmp(classLoadData->className, "java/lang/J9VMInternals") == 0) {
-		result = FALSE;
-	} else if (strcmp(classLoadData->className, "java/lang/invoke/VarHandleInternal") == 0) {
-		result = FALSE;
+canClassBeInstrumented(J9VMClassLoadHookEvent* data)
+{
+	BOOLEAN modifiable = TRUE;
+	/* Only clasess on the bootpath may be marked unmodifiable */
+	if (data->classLoader == data->currentThread->javaVM->systemClassLoader) {
+		U_8 *classData = data->classData;
+		UDATA classDataLength =  data->classDataLength;
+		U_8 const annotationLength = LITERAL_STRLEN(J9_UNMODIFIABLE_CLASS_ANNOTATION);
+		if (classDataLength > (annotationLength + 4)) {
+			/* Make sure this at least looks like a class file */
+			if ((classData[0] = 0xCA) && (classData[1] = 0xFE) && (classData[2] = 0xBA) && (classData[3] = 0xBE)) {
+				U_8 *dataEnd = classData + classDataLength - annotationLength;
+				classData += 4;
```

Comment:
This can skip ahead another 6 bytes {u2 major, u2 minor, u2 cp count}.

Reply:
Will roll this into the improved compare algorithm.